js/make_dead_fish_swim.js

```diff
@@ -0,0 +1,15 @@
+//https://www.codewars.com/kata/51e0007c1f9378fa810002a9/train/javascript
+
+function parse( data ){
+  const dataArray = data.split('');
+  const arrayReturn = [];
+  let element = 0;
+  dataArray.map(function(el){
```

Comment:
Why use `map`?

Reply:
Just training

Reply:
I mean why `map` and not sth like `forEach` :smile:

Reply:
I've added forEach soluton :)

Reply:
My intention was to replace `map` with `forEach`, as `map` is not necessary here :smile:

Reply:
I know, but I wanted to do this, especially with a map. So due to your suggestion, I've done that also with forEach.

Reply:
I think I don't understand.

* `map` should be used when you transform every element in some way and need an array of results.
* `forEach` is used when you want to invoke some logic for each element, but don't care about the array of results.

I would still suggest removing the solution with `map`, as seems unnecessary (not using the right tool for the job) :smile: